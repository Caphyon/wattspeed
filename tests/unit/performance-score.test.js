import Vue from "vue";
import { shallowMount } from "@vue/test-utils";
import speed_score from "../../src/components/panels/tech/items/performance-score.vue";

global.EventBus = new Vue();

describe("Testing speed-score comonent", () => {

  it("Should render the score of the page as danger", () => {
    const wrapper = shallowMount(speed_score, {
      propsData: {
        score: 0.14,
        type: "Desktop"
      },
    });
    // controll if the component is instance
    expect(wrapper.exists()).toBeTruthy();
    expect(wrapper.element).toMatchSnapshot();

    expect(wrapper.find('.text--center').exists()).toBe(true);
    expect(wrapper.find('.text--uppercase').exists()).toBe(true);
    expect(wrapper.find('.alert--danger').exists()).toBe(true);


  });

  it("Should render the score of the page as warning", () => {
    const wrapper = shallowMount(speed_score, {
      propsData: {
        score: 0.82,
        type: "Desktop"
      },
    });
    // controll if the component is instance
    expect(wrapper.exists()).toBeTruthy();
    expect(wrapper.element).toMatchSnapshot();

    expect(wrapper.find('.text--center').exists()).toBe(true);
    expect(wrapper.find('.text--uppercase').exists()).toBe(true);
    expect(wrapper.find('.alert--warning').exists()).toBe(true);


  });

  it("Should render the score of the page as success", () => {
    const wrapper = shallowMount(speed_score, {
      propsData: {
        score: 0.99,
        type: "Desktop"
      },
    });
    // controll if the component is instance
    expect(wrapper.exists()).toBeTruthy();
    expect(wrapper.element).toMatchSnapshot();

    expect(wrapper.find('.text--center').exists()).toBe(true);
    expect(wrapper.find('.text--uppercase').exists()).toBe(true);
    expect(wrapper.find('.alert--success').exists()).toBe(true);


  });

});
