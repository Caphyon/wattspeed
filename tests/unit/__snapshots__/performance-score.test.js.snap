// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing speed-score comonent Should render the score of the page as danger 1`] = `
<div
  class="text--center"
>
  <h3
    class="mt0 mb0 alert--danger"
  >
    14
  </h3>
   
  <span
    class="text--uppercase text--small"
  >
    Desktop
  </span>
</div>
`;

exports[`Testing speed-score comonent Should render the score of the page as success 1`] = `
<div
  class="text--center"
>
  <h3
    class="mt0 mb0 alert--success"
  >
    99
  </h3>
   
  <span
    class="text--uppercase text--small"
  >
    Desktop
  </span>
</div>
`;

exports[`Testing speed-score comonent Should render the score of the page as warning 1`] = `
<div
  class="text--center"
>
  <h3
    class="mt0 mb0 alert--warning"
  >
    82
  </h3>
   
  <span
    class="text--uppercase text--small"
  >
    Desktop
  </span>
</div>
`;
